src/self_driving/planning/pilot.cpp

```diff
@@ -32,27 +50,338 @@ Pilot::Pilot(std::string model_save_path, std::string forecast_model_save_path,
       settings_manager_(settings_manager),
       stats_storage_(stats_storage),
       txn_manager_(txn_manager),
-      workload_forecast_interval_(workload_forecast_interval) {
+      query_exec_util_(std::move(query_exec_util)),
+      task_manager_(task_manager),
+      workload_forecast_interval_(workload_forecast_interval),
+      sequence_length_(sequence_length),
+      horizon_length_(horizon_length) {
   forecast_ = nullptr;
   while (!model_server_manager_->ModelServerStarted()) {
   }
 }
 
+std::pair<uint64_t, uint64_t> Pilot::ComputeTimestampDataRange(uint64_t now) {
+  // Evaluation length is sequence length + 2 horizons
+  uint64_t eval_length = sequence_length_ + 2 * horizon_length_;
+
+  // Pull a range of the order of 5x (assuming classic 80% train/20% test split)
+  eval_length *= 5;
```

Comment:
This should only happen during training. During inference, we should just use `sequence_length_ + 2 * horizon_length_` amount of data.